src/Core/Grid/Definition/Factory/LogGridDefinitionFactory.php

```diff
@@ -0,0 +1,186 @@
+<?php
+/**
+ * 2007-2018 PrestaShop
+ *
+ * NOTICE OF LICENSE
+ *
+ * This source file is subject to the Open Software License (OSL 3.0)
+ * that is bundled with this package in the file LICENSE.txt.
+ * It is also available through the world-wide-web at this URL:
+ * https://opensource.org/licenses/OSL-3.0
+ * If you did not receive a copy of the license and are unable to
+ * obtain it through the world-wide-web, please send an email
+ * to license@prestashop.com so we can send you a copy immediately.
+ *
+ * DISCLAIMER
+ *
+ * Do not edit or add to this file if you wish to upgrade PrestaShop to newer
+ * versions in the future. If you wish to customize PrestaShop for your
+ * needs please refer to http://www.prestashop.com for more information.
+ *
+ * @author    PrestaShop SA <contact@prestashop.com>
+ * @copyright 2007-2018 PrestaShop SA
+ * @license   https://opensource.org/licenses/OSL-3.0 Open Software License (OSL 3.0)
+ * International Registered Trademark & Property of PrestaShop SA
+ */
+
+namespace PrestaShop\PrestaShop\Core\Grid\Definition\Factory;
+
+use PrestaShop\PrestaShop\Core\Grid\Action\GridAction;
+use PrestaShop\PrestaShop\Core\Grid\Action\GridActionCollection;
+use PrestaShop\PrestaShop\Core\Grid\Column\Column;
+use PrestaShop\PrestaShop\Core\Grid\Column\ColumnCollection;
+use Symfony\Component\Form\Extension\Core\Type\TextType;
+use Symfony\Component\Templating\EngineInterface;
+
+/**
+ * Class LogGridDefinitionFactory is responsible for creating new instance of Log grid definition
+ */
+final class LogGridDefinitionFactory extends AbstractGridDefinitionFactory
```

Comment:
@mickaelandrieu this class can define columns and other actions for grid and it can get quite big, do you think it's worth extracting them into separate classes? e.g. one class for providing grid columns, another class for row actions and so on.

Reply:
nope, for only 1 reason: we will create it using yaml in the end :p